csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -780,6 +783,62 @@ private static void HexEncodeUtf16CodeUnit(TextWriter writer, char c)
             writer.Write(Hex[(c >> 4) & 0xf]);
             writer.Write(Hex[(c >> 0) & 0xf]);
         }
+        
+        
+        private void WriteJsonObjectOpen(TextWriter writer, char openChar = '{')
+        {
+            writer.Write(openChar);
+            if (!settings.Pretty)
+            {
+                writer.Write(' ');
+            }
+        }
+
+        private void WriteJsonObjectClose(TextWriter writer, bool hasFields, int depth, char closeChar = '}')
+        {
+            if (hasFields)
+            {
+                if (settings.Pretty)
+                {
+                    writer.WriteLine();
+                    WriteIndentation(writer, depth);
+                }
+                else
+                {
+                    writer.Write(" ");
```

Comment:
The original implementation somewhat unconventionally added whitespace between braces `{ "key": "value" }`. Keeping the behavior to keep the contract.
For pretty printing, I did not add the spaces, as I thought it to be unconventional.